### PR TITLE
PTP support for l3ls_evpn

### PR DESCRIPTION
**Pull request by onurgashi**
## Change Summary

Add support for assigning `ptp` to p2p links and server profiles and ports.

P2P links exclude l2leaf and overlay-controller.
- Not required for overlay-controller
- Not all l2eaf switches support or require ptp

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Component(s) name
eos_l3ls_evpn

## Proposed changes

Add group_vars:

```
# Enable PTP on all infrastructure links | Optional
p2p_uplinks_ptp:
  enable: < boolean | default -> false >

port_profiles:
  < port_profile_1 >:
        # PTP Enable | Optional
        ptp:
          enable: < true | false >

servers:
  < server_1 >:
    adapters:
      - <...>
        # PTP Enable | Optional
        ptp:
          enable: < true | false >
```

Enabling ptp on server adapter will override ptp assigned under port_profile.

## How to test

https://github.com/onurgashi/claus-l3ls-dev.git

Test data:

```
p2p_uplinks_ptp:
  enable: true

port_profiles:
  TENANT_A_ALL:
    mode: trunk
    vlans: "120-130,160-161"
    ptp:
      enable: true

  server-2:
    rack: RackC
    adapters:
      - server_ports: [ Eth2]
        switch_ports: [ Ethernet9]
        switches: [ DC1-POD1-LEAF2A]
        profile: TENANT_A_ALL
        qos_profile: "bar"
        vlans: "124"
        mode: access
        ptp:
          enable: true
```

Verified structured config:

- [x] PTP on all infrastructure links
    - [x] super-spine - spine
    - [x] spine - leaf
- [x] Port profile with ptp var
- [x] Server-2 non-port-channel with ptp var
- [x] Server-2 with port-channel (missing feature in cli_config_gen)

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [ ] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md [hunk 1]
```diff
@@ -380,6 +380,9 @@ l2leaf_inband_management_vlan: < vlan_id >
 
 # QOS Profile assigned on all infrastructure links | Optional
 p2p_uplinks_qos_profile: < qos_profile_name >
+
+# Enable PTP on all infrastructure links | Optional
+p2p_uplinks_ptp_enable: < boolean | default -> false >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md]: please change to:
```yaml
p2p_uplinks_ptp:
  enable: <boolean | default -> false >
```
So we can add more knobs later

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/edge_ports/leaf-server-port-channels.j2 [hunk 2]
```diff
@@ -60,6 +61,12 @@
 {%                     if adapter_qos_profile is arista.avd.defined %}
     service_profile: {{ adapter_qos_profile }}
 {%                     endif %}
+{%                     if adapter_ptp is arista.avd.defined %}
+    ptp:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/edge_ports/leaf-server-port-channels.j2]: I don't think PTP is on port-channel level is it? I would think it was only for member ports.

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/spine-p2p-interfaces.j2 [hunk 3]
```diff
@@ -30,6 +30,10 @@
     mtu: {{ p2p_uplinks_mtu }}
 {%                 if p2p_uplinks_qos_profile is arista.avd.defined %}
     service_profile: {{ p2p_uplinks_qos_profile }}
+{%                 endif %}
+{%                 if p2p_uplinks_ptp_enable is arista.avd.defined %}
+    ptp:
+        enable: {{ p2p_uplinks_ptp_enable }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/spine-p2p-interfaces.j2]: only 2-space indentation in "structured configuration"

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/spine-p2p-interfaces.j2 [hunk 4]
```diff
@@ -77,6 +81,10 @@
     mtu: {{ p2p_uplinks_mtu }}
 {%              if p2p_uplinks_qos_profile is arista.avd.defined %}
     service_profile: {{ p2p_uplinks_qos_profile }}
+{%              endif %}
+{%              if p2p_uplinks_ptp_enable is arista.avd.defined %}
+    ptp:
+        enable: {{ p2p_uplinks_ptp_enable }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/spine-p2p-interfaces.j2]: only 2-space indentation in "structured configuration"

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/super-spine-p2p-interfaces.j2 [hunk 5]
```diff
@@ -24,6 +24,10 @@
     mtu: {{ p2p_uplinks_mtu }}
 {%             if p2p_uplinks_qos_profile is arista.avd.defined %}
     service_profile: {{ p2p_uplinks_qos_profile }}
+{%             endif %}
+{%             if p2p_uplinks_ptp_enable is arista.avd.defined %}
+    ptp:
+        enable: {{ p2p_uplinks_ptp_enable }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/super-spine-p2p-interfaces.j2]: only 2-space indentation in "structured configuration"

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/leaf-p2p-interfaces.j2 [hunk 6]
```diff
@@ -11,6 +11,10 @@
     mtu: {{ p2p_uplinks_mtu }}
 {%     if p2p_uplinks_qos_profile is arista.avd.defined %}
     service_profile: {{ p2p_uplinks_qos_profile }}
+{%     endif %}
+{%     if p2p_uplinks_ptp_enable is arista.avd.defined %}
+    ptp:
+        enable: {{ p2p_uplinks_ptp_enable }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/leaf-p2p-interfaces.j2]: only 2-space indentation in "structured configuration"

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md (hunk 1 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md]: Changed to p2p_uplinks_ptp

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/edge_ports/leaf-server-port-channels.j2 (hunk 2 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/edge_ports/leaf-server-port-channels.j2]: You are correct, not yet there with MLAG.

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/spine-p2p-interfaces.j2 (hunk 3 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/spine-p2p-interfaces.j2]: Changed, thanks!

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/spine-p2p-interfaces.j2 (hunk 4 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/spine-p2p-interfaces.j2]: Changed, thanks!

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/super-spine-p2p-interfaces.j2 (hunk 5 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/super-spine-p2p-interfaces.j2]: Changed, thanks!

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/leaf-p2p-interfaces.j2 (hunk 6 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/interfaces/leaf-p2p-interfaces.j2]: Changed, thanks!

**Review by ClausHolbechArista — APPROVED** [review]

**Comment by ClausHolbechArista** [comment]
LGTM

**Pull request merged**